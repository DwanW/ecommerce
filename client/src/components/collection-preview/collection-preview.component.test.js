import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPreview } from './collection-preview.component';

describe('Collection Preview component test', () => {
    let wrapper;
    let mockMatch;
    let mockHistory;
    const mockRouteName = 'hats';

    beforeEach(() => {
        mockMatch = {
            path: './shop'
        };
        mockHistory = {
            push: jest.fn()
        };

        const mockProps = {
            title: 'hats',
            items: [],
            history: mockHistory,
            match: mockMatch,
            routeName: mockRouteName
        };

        wrapper = shallow(<CollectionPreview {...mockProps} />);
    });

    it('should render CollectionPreview component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should fire router history function when clicked on titleContainer', () => {
        wrapper.find('TitleContainer').simulate('click');
        expect(mockHistory.push).toHaveBeenCalledWith(
            `${mockMatch.path}/${mockRouteName}`
        );
    });

    it('should render title prop in titleContainer', () => {
        expect(wrapper.find('TitleContainer').text()).toBe('HATS')
    });

    it('should render previewContainer with no child', () => {
        expect(wrapper.find('PreviewContainer').children().length).toBe(0);
    });
});