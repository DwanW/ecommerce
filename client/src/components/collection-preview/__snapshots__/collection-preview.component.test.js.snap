// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection Preview component test should render CollectionPreview component 1`] = `ShallowWrapper {}`;
